### PR TITLE
Upload private ssh key

### PR DESCRIPTION
**Pull request by vzhukovskii**
### What does this PR do?
This changes proposal add an ability to use theia api proposed in PR: https://github.com/eclipse-theia/theia/pull/6554 to upload private ssh key using new action item located in command list.

Demo of usage:
upload_ssh_key

Linked PR: https://github.com/eclipse-theia/theia/pull/6554

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14911

#### Release Notes

Add an ability to upload private SSH keys

### CODE REVIEW

**Comment by vzhukovskii** [comment]
CI might fails because changes were made in theia.d.ts in theia repository.

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 1]
```diff
@@ -237,6 +246,40 @@ const createKeyPair = async (sshkeyManager: SshKeyManager) => {
     }
 };
 
+const uploadPrivateKey = async (sshkeyManager: SshKeyManager) => {
+    let hostName = await getHostName('Please provide a hostname (e.g. github.com) or leave empty to setup default name');
+    if (!hostName) {
+        hostName = `default-${Date.now()}`;
+    }
```
Comment (vinokurig) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts]: I think we need to make this approach as a default for other methods e.g. create key, generate key.

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 2]
```diff
@@ -237,6 +246,40 @@ const createKeyPair = async (sshkeyManager: SshKeyManager) => {
...
+
+    const tempDir = mkdtempSync(join(os.tmpdir(), 'private-key-'));
+    const uploadedFilePaths = await theia.window.showUploadDialog({ defaultUri: theia.Uri.file(tempDir) });
+
+    if (!uploadedFilePaths || uploadPrivateKey.length === 0) {
+        theia.window.showErrorMessage('No private key uploaded');
```
Comment (vinokurig) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts]: `No private key has been uploaded` ?

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts [hunk 3]
```diff
@@ -237,6 +246,40 @@ const createKeyPair = async (sshkeyManager: SshKeyManager) => {
...
+
+    accessSync(privateKeyPath.path, R_OK);
+
+    const privateKeyContent = readFileSync(privateKeyPath.path).toString();
+
+    try {
+        await sshkeyManager.create({ name: hostName, service: 'vcs', privateKey: privateKeyContent });
+        await updateConfig(hostName);
+        await writeKey(hostName, privateKeyContent!);
+        theia.window.showInformationMessage(`Key pair for ${hostName} successfully created`);
```
Comment (vinokurig) [inline plugins/ssh-plugin/src/ssh-plugin-backend.ts]: `successfully uploaded` ?

Comment (vzhukovskii) [reply plugins/ssh-plugin/src/ssh-plugin-backend.ts]: Fiexed

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts (hunk 2 above)
Comment (vzhukovskii) [reply plugins/ssh-plugin/src/ssh-plugin-backend.ts]: Fixed

File: plugins/ssh-plugin/src/ssh-plugin-backend.ts (hunk 1 above)
Comment (vzhukovskii) [reply plugins/ssh-plugin/src/ssh-plugin-backend.ts]: Fixed

**Pull request merged**